client/src/use/useTrackFilters.ts

```diff
@@ -61,24 +67,27 @@ export default function useFilteredTracks(
   const filteredTracks = computed(() => {
     const checkedSet = new Set(checkedTypes.value);
     const confidenceThresh = defaultConfidenceThreshold.value;
-    return sortedTracks.value.filter((track) => {
-      const confidencePairsAboveThreshold = track.confidencePairs
-        .some(([confkey, confval]) => (
-          confval >= confidenceThresh && checkedSet.has(confkey)
-        ));
-      return (
+    const resultsArr: FilteredTrack[] = [];
+    sortedTracks.value.forEach((track) => {
+      const confidencePairIndex = track.confidencePairs
+        .findIndex(([confkey, confval]) => confval >= confidenceThresh && checkedSet.has(confkey));
         /* include tracks where at least 1 confidence pair is above
          * the threshold and part of the checked type set */
-        confidencePairsAboveThreshold
-        /* include tracks with no confidence pairs */
-        || track.confidencePairs.length === 0
-      );
+      if (confidencePairIndex !== -1 || track.confidencePairs.length === 0) {
```

Comment:
```suggestion
      if (confidencePairIndex >= 0 || track.confidencePairs.length === 0) {
```

Nit.  Include vs exclude condition.